sycl/include/CL/sycl/bit_cast.hpp

```diff
@@ -20,6 +22,10 @@ namespace sycl {
 // forward decl
 namespace detail {
 inline void memcpy(void *Dst, const void *Src, std::size_t Size);
+
+template <typename Type> struct type_helper { using T = Type; };
```

Comment:
can we rename the struct to something more explaining? lowering_type? or something which sounds better but also explains what it does.